app/models/application_contract.rb

```diff
@@ -0,0 +1,3 @@
+# frozen_string_literal: true
+
+ApplicationContract = Cinstance
```

Comment:
do you think this will always run after `cinstance`?

Reply:
That's not a problem in Rails 5, if `Cinstance` is a missing constant, the autoloader will find the file and load it. In Rails 6, I didn't find any mention to load order being an issue in the Zeitwerk docs, I assume that kind of problems are solved.